vs2022.imp

```diff
@@ -0,0 +1,10 @@
+[
```

Comment:
I'm not so sure about this mapping as a whole.

It's convenient for a VS2022 user, but it actually covers 3 unrelated libraries:

* The Windows SDK (which ships on a different cadence than Visual Studio)
* The MSVC STL (which I think is strictly speaking bound to the compiler release cadence, not the IDE... or maybe the compiler is bound to the IDE, not sure)
* The Universal CRT (which is Microsoft's attempt at a libc separate from compiler and IDE)

We typically ship mappings per library so they can be versioned as the library changes. It looks to me like these mappings are just what happened to work for your project, in which case it's probably better for you to maintain it as part of the project than as part of IWYU.

Let me know if I'm missing something there.

Reply:
Well, maybe that should be multiple mapping files then? I didn't do any special settings so these changes look like things that users of visual c++ are likely to stumble across.

Reply:
They look very sparse, I was curious how likely it is that another project using e.g. `"shlwapi.h"`, `<map>` or `<algorithm>` would work with these mappings? My guess is not very.

I've tried to build a library mapping generators under https://github.com/include-what-you-use/include-what-you-use/tree/master/mapgen so that users can generate mappings from the actual installed libraries rather than carrying hand-edited mappings for everything in the project. I think it would be better to invest in something like that for the Windows/MSVC libraries.

Reply:
I agree that this mapping looks very ad-hoc and tailored to a specific use case. I created a general issue about Windows.h / Windows SDK:
https://github.com/include-what-you-use/include-what-you-use/issues/1603

I agree with @kimgr that the three categories shouldn't share a single mapping.